### PR TITLE
Clippy Pedantic Acceptable Commits

### PR DESCRIPTION
**Pull request by dcampbell24**
These are the commits which you said you would accept.

https://github.com/uutils/coreutils/pull/6720

### CODE REVIEW

**Comment by dcampbell24** [comment]
^ [tertsdiepraam](https://github.com/tertsdiepraam)

File: src/uucore/src/lib/features/checksum.rs [hunk 1]
```diff
@@ -321,10 +322,10 @@ fn determine_regex(lines: &[String]) -> Option<(Regex, bool)> {
 
 // Converts bytes to a hexadecimal string
 fn bytes_to_hex(bytes: &[u8]) -> String {
-    bytes
-        .iter()
-        .map(|byte| format!("{byte:02x}"))
-        .collect::<String>()
+    bytes.iter().fold(String::new(), |mut bytes, byte| {
```
Comment (samueltardieu) [inline src/uucore/src/lib/features/checksum.rs:325]: Maybe create the string with `String::with_capacity(bytes.len() * 2)` first to ensure that there are no reallocations. Reallocations will typically appear as soon as there are more than 4 input bytes to format as the first allocated capacity for the string is usually 8 bytes.

Comment (BenWiederhake) [inline src/uucore/src/lib/features/checksum.rs:325]: Misleading variable name.

```suggestion
    bytes.iter().fold(String::new(), |mut hex, byte| {
```

**Review by BenWiederhake — APPROVED** [review]
LGTM, the changes are easily readable and clearly correspond to each clippy lint. This could be merged as-is. Bonus points if you also change the two small things that samueltardieu and I mentioned (preallocating the string and fixing a misleading variable name).

File: src/uucore/src/lib/features/checksum.rs (hunk 1 above)
Comment (dcampbell24) [reply src/uucore/src/lib/features/checksum.rs]: Did it.

**Comment by sylvestre** [comment]
could you please merge the two last commits into the relevant commit?

git rebase -i origin/main
will help
thanks

**Pull request closed without merging**